src/components/ChatEntry.js

```diff
@@ -1,22 +1,40 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './ChatEntry.css';
+import TimeStamp from './TimeStamp';
 import PropTypes from 'prop-types';
 
-const ChatEntry = (props) => {
+const ChatEntry = ( { sender, body, timeStamp, likeMessage, unlikeMessage } ) => {
+  const [hearted, setHearted] = useState(false);
+
+  const toggleHeart = () => {
+    setHearted(hearted => !hearted);
+    hearted ? unlikeMessage(): likeMessage();
+  };
+
+  const heartedIcon = hearted ? '❤️' : '🤍';
+
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{body}</p>
+        <p className="entry-time">
+          <TimeStamp time={timeStamp}/>
+        </p>
+        <button className="like" onClick={() => toggleHeart()}>
+          {heartedIcon}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number.isRequired,
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.number.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool.isRequired
```

Comment:
propTypes is missing a couple values that are passed in from ChatLog to ChatEntry: likeMessage and unlikeMessage